packages/web-components/src/components/va-combo-box/va-combo-box-library.js

```diff
@@ -0,0 +1,842 @@
+   */
+
+  /**
+   * Get an object of elements belonging directly to the given
+   * combo box component.
+   *
+   * @param {HTMLElement} el the element within the combo box
+   * @returns {ComboBoxContext} elements
+   */
+  const getComboBoxContext = el => {
+    const comboBoxEl = el.closest(COMBO_BOX);
+
+    if (!comboBoxEl) {
+      throw new Error(`Element is missing outer ${COMBO_BOX}`);
+    }
+
+    const selectEl = comboBoxEl.querySelector(SELECT);
+    const inputEl = comboBoxEl.querySelector(INPUT);
+    const listEl = comboBoxEl.querySelector(LIST);
+    const statusEl = comboBoxEl.querySelector(STATUS);
+    const focusedOptionEl = comboBoxEl.querySelector(LIST_OPTION_FOCUSED);
+    const selectedOptionEl = comboBoxEl.querySelector(LIST_OPTION_SELECTED);
+    const toggleListBtnEl = comboBoxEl.querySelector(TOGGLE_LIST_BUTTON);
+    const clearInputBtnEl = comboBoxEl.querySelector(CLEAR_INPUT_BUTTON);
+
+    const isPristine = comboBoxEl.classList.contains(COMBO_BOX_PRISTINE_CLASS);
+    const disableFiltering = comboBoxEl.dataset.disableFiltering === 'true';
+
+    return {
+      comboBoxEl,
+      selectEl,
+      inputEl,
+      listEl,
+      statusEl,
+      focusedOptionEl,
+      selectedOptionEl,
+      toggleListBtnEl,
+      clearInputBtnEl,
+      isPristine,
+      disableFiltering,
+    };
+  };
+
+  /**
+   * Disable the combo-box component
+   *
+   * @param {HTMLInputElement} el An element within the combo box component
+   */
+  const disable = el => {
+    const { inputEl, toggleListBtnEl, clearInputBtnEl } =
+      getComboBoxContext(el);
+
+    clearInputBtnEl.hidden = true;
+    clearInputBtnEl.disabled = true;
+    toggleListBtnEl.disabled = true;
+    inputEl.disabled = true;
+  };
+
+  /**
+   * Check for aria-disabled on initialization
+   *
+   * @param {HTMLInputElement} el An element within the combo box component
+   */
+  const ariaDisable = el => {
+    const { inputEl, toggleListBtnEl, clearInputBtnEl } =
+      getComboBoxContext(el);
+
+    clearInputBtnEl.hidden = true;
+    clearInputBtnEl.setAttribute('aria-disabled', true);
+    toggleListBtnEl.setAttribute('aria-disabled', true);
+    inputEl.setAttribute('aria-disabled', true);
+  };
+
+  /**
+   * Enable the combo-box component
+   *
+   * @param {HTMLInputElement} el An element within the combo box component
+   */
+  const enable = el => {
+    const { inputEl, toggleListBtnEl, clearInputBtnEl } =
+      getComboBoxContext(el);
+
+    clearInputBtnEl.hidden = false;
+    clearInputBtnEl.disabled = false;
+    toggleListBtnEl.disabled = false;
+    inputEl.disabled = false;
+  };
+
+  /**
+   * Enhance a select element into a combo box component.
+   *
+   * @param {HTMLElement} _comboBoxEl The initial element of the combo box component
+   */
+  const enhanceComboBox = (_comboBoxEl, _labelEl) => {
+    const comboBoxEl = _comboBoxEl.closest(COMBO_BOX);
+    console.log('comboBoxEl', _comboBoxEl);
+    console.log('_labelEl', _labelEl);
```

Comment:
Do you mind cleaning up these console.logs before merging? They are printing in the test runs `yarn test`.

![Screenshot 2024-11-21 at 9 51 41 AM](https://github.com/user-attachments/assets/58455939-607e-4eb2-9341-c8fa76307d45)

Reply:
yikes...